### PR TITLE
189 notes on importing data

### PR DESCRIPTION
**Pull request by daveatweaviate**
[issue](https://github.com/weaviate/weaviate-edu-team-tasks/issues/189)
[staging](https://189-notes-on-importing-data--tangerine-buttercream-20c32f.netlify.app/developers/weaviate)

### CODE REVIEW

File: developers/weaviate/concepts/cluster.md [hunk 1]
```diff
@@ -47,7 +47,9 @@ All of the scenarios below assume that - as sharding or replication is increased
 
 ### Advantages when increasing sharding
 * Run larger datasets
-* Speed up imports
+* Speed up imports.
+
+To use multiple CPUs efficiently, enable sharding when you import data. For the fastest imports, enable sharding even on a single node.
```
Comment (databyjp) [inline developers/weaviate/concepts/cluster.md:52]: I think sharding is always "on", as each collection always includes one or more shards. So the terms "enable sharding" might be confusing. Propose to remove this sentence.

File: developers/weaviate/concepts/resources.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ The CPU has a direct effect on query and import speed, but does not affect datas
 
 Vector search is the most CPU intensive process in Weaviate operations. Queries are CPU-bound, but imports are also CPU-bound because imports rely on vector search for indexing. Weaviate uses the HNSW (Hierarchical Navigable Small World) algorithm to index vectors. You can [tune the HNSW index](../config-refs/schema/vector-index.md) on a per collection basis in order to maximize performance for your primary use case.
 
+To use multiple CPUs efficiently, enable sharding when you import data. For the fastest imports, enable sharding even on a single node.
```
Comment (databyjp) [inline developers/weaviate/concepts/resources.md:35]: Propose to remove this for the same reason.

File: developers/weaviate/concepts/resources.md [hunk 3]
```diff
@@ -87,6 +89,9 @@ The memory calculation that includes `maxConnections` describes the system state
 ### Out-of-Memory issues due to garbage collection
 In rare situations - typically on large machines with very high import speeds - Weaviate can allocate memory faster than the garbage collector can free it. When this happens, the system kernel can trigger an `out of memory kill (OOM-Kill)`. This is a known issue that Weaviate is actively working on.
 
+### Data import
+To avoid out-of-memory issues during imports, configure the GOMEMLIMIT environment variable. For details, see [Environment variables](../config-refs/env-vars), [blog post](/blog/weaviate-1-15-release#better-control-over-garbage-collector).
```
Comment (databyjp) [inline developers/weaviate/concepts/resources.md:93]: Is this still valid given the new resource limit config available from `1.23`?

File: developers/weaviate/concepts/storage.md [hunk 4]
```diff
@@ -59,7 +59,11 @@ By grouping a vector index with the object storage within a shard, Weaviate can
 
 ### Shard Components Optimizations
 
-As a rule of thumb: For structured/object data, Weaviate's storage mechanisms accept and embrace segmentation as segments are cheap to merge and even unmerged segments can be navigated efficiently thanks to Bloom filters. In turn, the ingestion speed is high and does not degrade over time. For the vector index, Weaviate aims to keep the index as large as possible within a shard, as HNSW indices cannot efficiently be merged and querying a single large index is more efficient that sequentially querying many small indices.
+Weaviate's storage mechanisms use segmentation for structured/object data. Segments are cheap to merge and even unmerged segments can be navigated efficiently thanks to Bloom filters. In turn, ingestion speed is high and does not degrade over time.
+
+Weaviate keeps the vector index as large as possible within a shard. HNSW indices cannot be merged efficiently. Querying a single large index is more efficient than sequentially querying many small indexes.
+
+To use multiple CPUs efficiently, enable sharding when you import data. For the fastest imports, enable sharding even on a single node.
```
Comment (databyjp) [inline developers/weaviate/concepts/storage.md:66]: Same comment as above.

File: developers/weaviate/concepts/cluster.md (hunk 1 above)
Comment (daveatweaviate) [reply developers/weaviate/concepts/cluster.md:52]: I think this means you need to have multiple shards to take advantage of multiple CPUs. I'll ask for clarification.

**Comment by daveatweaviate** [comment]
Waiting on feedback about multiple shards

**Pull request merged**